Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift

```diff
@@ -2,417 +2,93 @@
 //  SPDX-License-Identifier: Apache-2.0.
 
 import struct Foundation.Data
-import class Foundation.InputStream
-import class Foundation.FileManager
-import class Foundation.FileHandle
-import class Foundation.OutputStream
-import struct Foundation.URL
 import AwsCIo
-import AwsCCommon
 import AwsCCal
-#if os(Linux)
-     import Glibc
- #else
-     import Darwin
- #endif
 
-// swiftlint:disable identifier_name superfluous_disable_command
-// Todo: this is being used. Refactor
-public class ByteBuffer: Codable {
+public final class ByteBuffer: Decodable {
+
+    private var data: Data
+    private var currentIndex: Data.Index
 
     public init(size: Int) {
-        array.reserveCapacity(size)
-        self.capacity = size
+        data = Data(capacity: size)
+        currentIndex = data.startIndex
     }
 
     public init(bytes: [UInt8]) {
-        self.array = bytes
-        self.capacity = bytes.count
-    }
-
-    //Todo: do not expose pointer
-    public init(ptr: UnsafeMutablePointer<UInt8>, len: Int, capacity: Int) {
-        let buffer = UnsafeBufferPointer(start: ptr, count: len)
-        self.array = Array(buffer)
-        self.capacity = capacity
-    }
-
-    public func allocate(_ size: Int) {
-        array = [UInt8]()
-        array.reserveCapacity(size)
-        self.capacity += size
-        currentIndex = 0
-    }
-
-    public func nativeByteOrder() -> Endianness {
-        return hostEndianness
+        self.data = Data(bytes)
+        currentIndex = data.startIndex
     }
 
-    public func currentByteOrder() -> Endianness {
-        return currentEndianness
+    public init(data: Data) {
+        self.data = data
+        currentIndex = data.startIndex
     }
 
-    public func order(_ endianness: Endianness) -> ByteBuffer {
-        currentEndianness = endianness
-        return self
+    public init(bufferPointer: UnsafeMutablePointer<UInt8>, length: Int, capacity: Int) {
+        self.data = Data(bytes: bufferPointer, count: capacity)
+        currentIndex = data.startIndex
     }
 
-    public func putByte(_ value: UInt8) {
-        array.append(value)
+    public init(from decoder: Decoder) throws {
+        data = try Data(from: decoder)
+        currentIndex = data.startIndex
     }
 
-    public func put(_ value: UInt8) -> ByteBuffer {
-        array.append(value)
-        return self
+    public func put(_ value: UInt8) {
+        data.append(value)
     }
 
-    public func put(_ value: [UInt8]) -> ByteBuffer {
-        array.append(contentsOf: value)
-        return self
-    }
-
-    public func put(_ value: ByteBuffer, offset: UInt = 0, maxBytes: UInt? = nil) {
-        var end: UInt = UInt(value.length())
-        if let maxBytes = maxBytes {
-             end = maxBytes
-        }
-        let bytesToTake = value.array[Int(offset)..<Int(end)]
-        array.append(contentsOf: bytesToTake)
+    public func put(_ value: [UInt8]) {
```

Comment:
let's take these put() functions out if no one is using them
red code is best code

Reply:
Thanks, nuked.